flow-typed/redux.js

```diff
@@ -0,0 +1,74 @@
+  };
+
+  declare type StoreEnhancer<S, A> = (next: StoreCreator<S, A>) => StoreCreator<S, A>;
+
+  declare type Fn0<A> = () => A;
+  declare type Fn1<A, B1> = (b1: B1) => A;
+  declare type Fn2<A, B1, B2> = (b1: B1, b2: B2) => A;
+  declare type Fn3<A, B1, B2, B3> = (b1: B1, b2: B2, b3: B3) => A;
+  declare type Fn4<A, B1, B2, B3, B4> = (b1: B1, b2: B2, b3: B3, b4: B4) => A;
+  declare type Fn5<A, B1, B2, B3, B4, B5> = (b1: B1, b2: B2, b3: B3, b4: B4, b5: B5) => A;
+  declare type Fn6<A, B1, B2, B3, B4, B5, B6> = (b1: B1, b2: B2, b3: B3, b4: B4, b5: B5, b6: B6) => A;
+  declare type Fn7<A, B1, B2, B3, B4, B5, B6, B7> = (b1: B1, b2: B2, b3: B3, b4: B4, b5: B5, b6: B6, b7: B7) => A;
+  declare type Fn8<A, B1, B2, B3, B4, B5, B6, B7, B8> = (b1: B1, b2: B2, b3: B3, b4: B4, b5: B5, b6: B6, b7: B7, b8: B8) => A;
+  declare type Fn9<A, B1, B2, B3, B4, B5, B6, B7, B8, B9> = (b1: B1, b2: B2, b3: B3, b4: B4, b5: B5, b6: B6, b7: B7, b8: B8, b9: B9) => A;
+
+  declare type ActionCreators<K, A> = { [key: K]: (...args: Array<any>) => A };
+
+  declare function createStore<S, A>(reducer: Reducer<S, A>, enhancer?: StoreEnhancer<S, A>): Store<S, A>;
+  declare function createStore<S, A>(reducer: Reducer<S, A>, preloadedState: S, enhancer?: StoreEnhancer<S, A>): Store<S, A>;
+
+  declare function applyMiddleware<S, A>(...middlewares: Array<Middleware<S, A>>): StoreEnhancer<S, A>;
+
+  declare function bindActionCreators<A>(actionCreators: Fn0<A>, dispatch: Dispatch<A>): Fn0<A>;
+  declare function bindActionCreators<A, B1>(actionCreators: Fn1<A, B1>, dispatch: Dispatch<A>): Fn1<A, B1>;
+  declare function bindActionCreators<A, B1, B2>(actionCreators: Fn2<A, B1, B2>, dispatch: Dispatch<A>): Fn2<A, B1, B2>;
+  declare function bindActionCreators<A, B1, B2, B3>(actionCreators: Fn3<A, B1, B2, B3>, dispatch: Dispatch<A>): Fn3<A, B1, B2, B3>;
+  declare function bindActionCreators<A, B1, B2, B3, B4>(actionCreators: Fn4<A, B1, B2, B3, B4>, dispatch: Dispatch<A>): Fn4<A, B1, B2, B3, B4>;
+  declare function bindActionCreators<A, B1, B2, B3, B4, B5>(actionCreators: Fn5<A, B1, B2, B3, B4, B5>, dispatch: Dispatch<A>): Fn5<A, B1, B2, B3, B4, B5>;
+  declare function bindActionCreators<A, B1, B2, B3, B4, B5, B6>(actionCreators: Fn6<A, B1, B2, B3, B4, B5, B6>, dispatch: Dispatch<A>): Fn6<A, B1, B2, B3, B4, B5, B6>;
+  declare function bindActionCreators<A, B1, B2, B3, B4, B5, B6, B7>(actionCreators: Fn7<A, B1, B2, B3, B4, B5, B6, B7>, dispatch: Dispatch<A>): Fn7<A, B1, B2, B3, B4, B5, B6, B7>;
+  declare function bindActionCreators<A, B1, B2, B3, B4, B5, B6, B7, B8>(actionCreators: Fn8<A, B1, B2, B3, B4, B5, B6, B7, B8>, dispatch: Dispatch<A>): Fn8<A, B1, B2, B3, B4, B5, B6, B7, B8>;
+  declare function bindActionCreators<A, B1, B2, B3, B4, B5, B6, B7, B8, B9>(actionCreators: Fn9<A, B1, B2, B3, B4, B5, B6, B7, B8, B9>, dispatch: Dispatch<A>): Fn9<A, B1, B2, B3, B4, B5, B6, B7, B8, B9>;
```

Comment:
So something like

```js
type ActionCreator<A> = (...args: Array<any>) => A;
type ActionCreators<K, A> = { [key: K]: ActionCreator<A> };

declare function bindActionCreators<A, K, C: ActionCreator<A>>(actionCreators: C, dispatch: Dispatch<A>): C;
declare function bindActionCreators<A, K, C: ActionCreators<K, A>>(actionCreators: C, dispatch: Dispatch<A>): C;
```

right? It's a great improvement, thanks a lot @aaronjensen